aspnet/fundamentals/servers.rst

```diff
@@ -1,8 +1,274 @@
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpSendFileFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IHttpWebSocketFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - IRequestIdentifierFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - ISessionFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - ITlsConnectionFeature
+	  - Yes
+	  - Yes
+	  - No
+	* - ITlsTokenBindingFeature
+	  - Yes
+	  - Yes
+	  - No
+
+For the most part, feature interfaces are specified on the request object. We'll see below how `Kestrel`_ implements the features it supports.
+
+Configuration options
+^^^^^^^^^^^^^^^^^^^^^
+
+When launching a server, you can provide it with some configuration options. This can be done directly using command line parameters, or a configuration file containing the settings can be specified. The ``Microsoft.AspNet.Hosting`` command supports parameters for the server to use (such as ``Kestrel`` or ``WebListener``) as well as a ``server.urls`` configuration key, which should contain a semicolon-separated list of URL prefixes the server should handle.
+
+The ``project.json`` file shown above demonstrates how to pass the ``server.urls`` parameter directly:
+
+.. code-block:: javascript
+
+	"kestrel": "Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5004"
+
+Alternately, a configuration file can be referenced, instead:
+
+.. code-block:: javascript
+
+	"kestrel": "Microsoft.AspNet.Hosting --config hosting.ini"
+
+Then, ``hosting.ini`` can include the settings the server will use (including the server parameter, as well):
+
+.. code-block:: text
+
+	server=Kestrel
+	server.urls=http://localhost:5004
+
+Programmatic configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to specifying configuration options, the server hosting the application can be referenced programmatically via the `IApplicationBuilder interface <https://github.com/aspnet/HttpAbstractions/blob/ed339a35d2a0ae6137c12e9bc8e8b037ed429bc1/src/Microsoft.AspNet.Http.Abstractions/IApplicationBuilder.cs>`_, available in the ``Configure`` method in ``Startup``. ``IApplicationBuilder`` exposes a Server property of type `ServerInformation <https://github.com/aspnet/Hosting/blob/62e534977386ebb02bfc351ec4a6bbea9ea7f635/src/Microsoft.AspNet.Hosting.Server.Abstractions/IServerInformation.cs>`_. ``IServerInformation`` only exposes a ``Name`` property, but different server implementations may expose additional functionality. For instance, WebListener exposes a ``Listener`` property that can be used to configure the server's authentication:
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/Startup.cs
+	:linenos:
+	:lines: 29-45
+	:emphasize-lines: 3,6-8,13
+	:language: c#
+	:dedent: 8
+
+In addition to configuring the server, you can also adjust how your application will behave based on which server is hosting it, as provided by ``IServerInformation.Name`` (as shown on line 13 in the code listing above). The following table lists the servers and the name they provide:
```

Comment:
I'll remove that section. Out of curiosity, do you see a use case other than logging in which IServerInformation.Name should be used, then?